packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -1209,15 +1211,15 @@ class SemanticsNode extends AbstractNode with DiagnosticableTreeMixin {
   ///
   /// Keys are used during the construction of the semantics tree. They are not
   /// transferred to the engine.
-  final Key key;
+  final Key? key;
```

Comment:
(unrelated to this PR)

/cc @chunhtai do you know what we use the key of a SemanticsNode for? I did a quick search and I couldn't figure out why we have this...

Reply:
It is used by custom painter to maintains its semantics sub tree i think

Reply:
The key is used by CustomPainterSemantics to identify whether a semantics node can be update by the CustomPainterSemantics

Reply:
Ah, makes sense!